src/sys/windows/ll.rs

```diff
@@ -0,0 +1,219 @@
+}
+
+impl<T: Send + Sync> LinkedList<T> {
+    pub fn new() -> LinkedList<T> {
+        LinkedList {
+            head: None,
+            tail: Rawlink::none(),
+        }
+    }
+}
+
+impl<T> LinkedList<T> {
+    pub fn is_empty(&self) -> bool {
+        self.head.is_none()
+    }
+
+    pub fn get_mut(&mut self, entry: &Entry<T>) -> &mut T {
+        self.ensure_same_ll(&entry);
+
+        unsafe {
+            // Save because &mut self has mutable access to the entire
+            // LinkedList
+            let n: Option<&mut Node<T>> = mem::transmute(entry.node);
+            &mut n.unwrap().value
+        }
+    }
+
+    pub fn push(&mut self, el: T) -> Entry<T> {
+        let mut node = Box::new(Node::new(el));
+        let entry = Entry::new(self, &mut node);
+
+        match unsafe { self.tail.resolve_mut() } {
+            None => self.push_front(node),
+            Some(tail) => {
+                tail.set_next(node);
+                self.tail = Rawlink::from(&mut tail.next);
+            }
+        }
+
+        entry
+    }
+
+    pub fn remove(&mut self, mut entry: Entry<T>) -> T {
+        self.ensure_same_ll(&entry);
+
+        let (mut prev, next) = unsafe {
+            let node = entry.node.resolve_mut().expect("invalid entry");
+            (node.prev, node.next.take())
+        };
+
+        // Unlink previous pointer
+        let removed = match unsafe { (prev.resolve_mut(), next) } {
+            (Some(p), Some(mut next)) => {
+                next.prev = prev;
+                *mem::replace(&mut p.next, Some(next)).take().unwrap()
+            }
+            (None, Some(mut next)) => {
+                next.prev = Rawlink::none();
+                *mem::replace(&mut self.head, Some(next)).take().unwrap()
+            }
+            (Some(p), None) => {
+                self.tail = prev;
+                *p.next.take().unwrap()
+            }
+            (None, None) => {
+                self.tail = Rawlink::none();
+                *self.head.take().unwrap()
+            }
+        };
+
+        debug_assert!(removed.next.is_none());
+        removed.value
+    }
+
+    pub fn iter(&self) -> Iter<T> {
+        Iter { curr: &self.head }
+    }
+
+    fn push_front(&mut self, mut new_head: Box<Node<T>>) {
+        match self.head {
+            None => {
+                self.head = link_no_prev(new_head);
+                self.tail = Rawlink::from(&mut self.head);
+            }
+            Some(ref mut head) => {
+                new_head.prev = Rawlink::none();
+                head.prev = Rawlink::some(&mut *new_head);
+                mem::swap(head, &mut new_head);
+                head.next = Some(new_head);
+            }
+        }
+    }
+
+    fn ensure_same_ll(&self, entry: &Entry<T>) {
+        assert!(entry.ll == self as *const LinkedList<T> as *mut LinkedList<T>, "entry belongs to a different LinkedList");
+    }
+}
+
+unsafe impl<T> Send for LinkedList<T> {}
+
+pub struct Entry<T> {
```

Comment:
The lack of a lifetime connection between `Entry` and `LinkedList` makes me somewhat uneasy here, and the blind `unsafe impl<T> Send for LinkedList<T>` is also somewhat worrying (e.g. no bounds on `T`). I wonder if it would perhaps be better to just start out with a `HashMap` instead of a linked list? That should in theory still have `O(1)` lookup, addition, and removal (just with some higher constants out in front), but the perf for the level implementation could always be improved later on.